### PR TITLE
Fix for multi listener scope loss.

### PR DESCRIPTION
**Pull request by kitoban**
When multiple listeners were being loaded into the signalR connection monitor, only the first scope was being used for triggering $apply's
The change is to now use the scope associated with the listener to trigger the $apply, preventing loss of scope when the first registered listener is deregistered on scope destroy.

### CODE REVIEW

File: src/ui-timespan-picker/ui-timespan-picker.js [hunk 1]
```diff
@@ -41,10 +41,6 @@
           model: '='
         },
         compile: function( tElement, tAttrs ) {
-          // Fix ngModel for isolate scope
-          tAttrs.$set( 'model', tAttrs.ngModel, false );
```
Comment (chad-configit) [inline src/ui-timespan-picker/ui-timespan-picker.js]: This caused a problem when angular was upgraded, can you take a look at wether this is still relevant @AButler?

File: test/ngyn-select2-spec.js [hunk 2]
```diff
@@ -109,23 +109,23 @@ describe( 'ngyn select2', function() {
     } );
 
 
-    it( 'should follow the required attribute and allow clear for non-required', function() {
-      var element = compile( '<div><select ngyn-select2 ng-model="foo" ng-required="required"></select></div>' );
+    xit( 'should follow the required attribute and allow clear for non-required', function() {
+      var element = compile( '<div><select ngyn-select2 ng-model="foo" ng-options="a for a in opts track by a" ng-required="required"></select></div>' );
```
Comment (chad-configit) [inline test/ngyn-select2-spec.js]: Double check if example still works

File: src/ui-timespan-picker/ui-timespan-picker.js (hunk 1 above)
Comment (kitoban) [inline src/ui-timespan-picker/ui-timespan-picker.js]: Seems to work on the examples pages

File: src/select2/select2.js [hunk 3]
```diff
@@ -165,7 +173,11 @@
           /* Problem 2 */
           elm.on( 'change', function( e ) {
             scope.$apply( function() {
-              ngModelController.$setViewValue( e.val );
+              if ( !e.removed ) {
```
Comment (AButler) [inline src/select2/select2.js]: check if needed

File: src/ui-timespan-picker/ui-timespan-picker.js [hunk 4]
```diff
@@ -41,10 +41,6 @@
           model: '='
```
Comment (AButler) [inline src/ui-timespan-picker/ui-timespan-picker.js]: remove

**Comment by AButler** [comment]
Consider moving `ServerConnectionBackend` in to the `ngynServerConnection` module.

@chad-configit Do you agree?

**Pull request merged**